### PR TITLE
Historical graph time range

### PR DESCRIPTION
**Pull request by treyyoder**
- Added a range selector for the historical time line (6 months, 1 year, 5 years, all time).
- Moved the historical timeline to below the header of the detail page.

### CODE REVIEW

File: imports/startup/server/index.coffee [hunk 1]
```diff
@@ -313,27 +313,24 @@ api.addRoute 'frequentAgents',
...
+        date.subtract(6, 'months')
+        dateStr = date.format("YYYY-MM-DD") + "T00:00:00+00:01"
+      when "1year"
+        date.subtract(1, 'years')
+        dateStr = date.format("YYYY-MM-DD") + "T00:00:00+00:01"
+      when "5years"
+        date.subtract(5, 'years')
+        dateStr = date.format("YYYY") + "-01-01T00:00:00+00:01"
+      when "all"
+        date.subtract(100, 'years')
```
Comment (nathanathan) [inline imports/startup/server/index.coffee]: Why is this offset needed? It looks like the filter isn't used when the date range is "all."

File: imports/startup/server/index.coffee [hunk 2]
```diff
@@ -348,14 +345,27 @@ api.addRoute 'historicalData/:term',
...
-      }
-      GROUP BY ?word ?year
       """
+    if @urlParams.range != 'all'
+      query += """
+        FILTER (?dateTime > "#{escape(dateStr)}"^^xsd:dateTime)
+        """
+    if @urlParams.range == '6months' || @urlParams.range == '1year'
+      query += """
+        BIND(month(?dateTime) AS ?year)
```
Comment (nathanathan) [inline imports/startup/server/index.coffee]: The "year" variable should be renamed to something like timeInterval.

File: imports/ui/components/timeline.coffee [hunk 3]
```diff
@@ -1,46 +1,117 @@
...
+            baseYear++
+        when "all"
+          baseYear = @tld.find({}, {sort: {year: 1}}).fetch()[0].year
+          while baseYear <= endDate.year()
+            tdata = @tld.find({year: baseYear}).fetch()
+            if tdata.length == 0
+              counts.push 0
+            else
+              if tdata[0].count > maxCount
+                maxCount = tdata[0].count
```
Comment (nathanathan) [inline imports/ui/components/timeline.coffee]: I would suggest computing this after the counts have all been inserted with `_.max(counts)` so it doesn't need to be repeated in each condition.

**Comment by nathanathan** [comment]
Could you add a blue divider below the plot like the mockup here: https://drive.google.com/a/ecohealthalliance.org/file/d/0B2aeANKs4QcPTkRxdFFZdEhUWG8/view

**Comment by treyyoder** [comment]
@nathanathan: This is ready for review.

**Comment by nathanathan** [comment]
👍

**Pull request merged**